prow/plugins/project/project.go

```diff
@@ -325,10 +327,19 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, p
 	}
 
 	projectCard := github.ProjectCard{}
-	projectCard.ContentID = e.Number
```

Comment:
Are you sure this is not in the event payload? I would expect it to be.

Reply:
if you are referring to GenericCommentEvent, it does not contain the information I need. Also please note that ID is added in this PR.

Reply:
Is it in the webhook payloads from GitHub, though? Can we add it to the GenericCommentEvent?

Reply:
AFAIK only url of the issue/pr is in the web hook payload. We still need a api call to get the ID. API call seems inevitable.

Reply:
In the following issue comment creation payload, I see the ID at `.issue.id`. Which types of events are you seeing that not present in?

```json
{
  "action": "created",
  "issue": {
    "url": "https://api.github.com/repos/openshift/origin/issues/22852",
    "repository_url": "https://api.github.com/repos/openshift/origin",
    "labels_url": "https://api.github.com/repos/openshift/origin/issues/22852/labels{/name}",
    "comments_url": "https://api.github.com/repos/openshift/origin/issues/22852/comments",
    "events_url": "https://api.github.com/repos/openshift/origin/issues/22852/events",
    "html_url": "https://github.com/openshift/origin/pull/22852",
    "id": 445236121,
    "node_id": "MDExOlB1bGxSZXF1ZXN0Mjc5NzI5NzQ5",
    "number": 22852,
    "title": "Bug 1709958: test/e2e/upgrade: Add ingress controller test",
    "user": {
      "login": "Miciah",
      "id": 2042465,
      "node_id": "MDQ6VXNlcjIwNDI0NjU=",
      "avatar_url": "https://avatars1.githubusercontent.com/u/2042465?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/Miciah",
      "html_url": "https://github.com/Miciah",
      "followers_url": "https://api.github.com/users/Miciah/followers",
      "following_url": "https://api.github.com/users/Miciah/following{/other_user}",
      "gists_url": "https://api.github.com/users/Miciah/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/Miciah/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/Miciah/subscriptions",
      "organizations_url": "https://api.github.com/users/Miciah/orgs",
      "repos_url": "https://api.github.com/users/Miciah/repos",
      "events_url": "https://api.github.com/users/Miciah/events{/privacy}",
      "received_events_url": "https://api.github.com/users/Miciah/received_events",
      "type": "User",
      "site_admin": false
    },
    "labels": [
      {
        "id": 640548394,
        "node_id": "MDU6TGFiZWw2NDA1NDgzOTQ=",
        "url": "https://api.github.com/repos/openshift/origin/labels/size/L",
        "name": "size/L",
        "color": "ee9900",
        "default": false
      }
    ],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [

    ],
    "milestone": null,
    "comments": 2,
    "created_at": "2019-05-17T02:25:50Z",
    "updated_at": "2019-05-21T00:05:26Z",
    "closed_at": null,
    "author_association": "CONTRIBUTOR",
    "pull_request": {
      "url": "https://api.github.com/repos/openshift/origin/pulls/22852",
      "html_url": "https://github.com/openshift/origin/pull/22852",
      "diff_url": "https://github.com/openshift/origin/pull/22852.diff",
      "patch_url": "https://github.com/openshift/origin/pull/22852.patch"
    },
    "body": "The test fails if the route is unavailable before or after an upgrade.  For now, the test only logs if the test is unavailable during an upgrade.\r\n\r\n* `test/e2e/upgrade/ingress_controller.go`: New file.  Test that the default ingress controller is available before, during, and after an upgrade.\r\n* `test/e2e/upgrade/upgrade.go` (`AllTests`): Add `IngressControllerUpgradeTest`.\r\n"
  },
  "comment": {
    "url": "https://api.github.com/repos/openshift/origin/issues/comments/494191807",
    "html_url": "https://github.com/openshift/origin/pull/22852#issuecomment-494191807",
    "issue_url": "https://api.github.com/repos/openshift/origin/issues/22852",
    "id": 494191807,
    "node_id": "MDEyOklzc3VlQ29tbWVudDQ5NDE5MTgwNw==",
    "user": {
      "login": "openshift-ci-robot",
      "id": 30189103,
      "node_id": "MDQ6VXNlcjMwMTg5MTAz",
      "avatar_url": "https://avatars1.githubusercontent.com/u/30189103?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/openshift-ci-robot",
      "html_url": "https://github.com/openshift-ci-robot",
      "followers_url": "https://api.github.com/users/openshift-ci-robot/followers",
      "following_url": "https://api.github.com/users/openshift-ci-robot/following{/other_user}",
      "gists_url": "https://api.github.com/users/openshift-ci-robot/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/openshift-ci-robot/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/openshift-ci-robot/subscriptions",
      "organizations_url": "https://api.github.com/users/openshift-ci-robot/orgs",
      "repos_url": "https://api.github.com/users/openshift-ci-robot/repos",
      "events_url": "https://api.github.com/users/openshift-ci-robot/events{/privacy}",
      "received_events_url": "https://api.github.com/users/openshift-ci-robot/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2019-05-21T00:05:26Z",
    "updated_at": "2019-05-21T00:05:26Z",
    "author_association": "NONE",
    "body": "@Miciah: The following test **failed**, say `/retest` to rerun them all:\n\nTest name | Commit | Details | Rerun command\n--- | --- | --- | ---\nci/prow/e2e-aws-upgrade | b65e63f3804b8de0b1b0478ed0c1922adb3c241c | [link](https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/22852/pull-ci-openshift-origin-master-e2e-aws-upgrade/1008) | `/test e2e-aws-upgrade`\n\n[Full PR test history](https://openshift-gce-devel.appspot.com/pr/22852). [Your PR dashboard](https://openshift-gce-devel.appspot.com/pr/Miciah). Please help us cut down on flakes by [linking to](https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/openshift/origin/issues?q=is:issue+is:open) when you hit one in your PR.\n\n<details>\n\nInstructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).\n</details>\n<!-- test report -->"
  },
  "repository": {
    "id": 22442668,
    "node_id": "MDEwOlJlcG9zaXRvcnkyMjQ0MjY2OA==",
    "name": "origin",
    "full_name": "openshift/origin",
    "private": false,
    "owner": {
      "login": "openshift",
      "id": 792337,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjc5MjMzNw==",
      "avatar_url": "https://avatars3.githubusercontent.com/u/792337?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/openshift",
      "html_url": "https://github.com/openshift",
      "followers_url": "https://api.github.com/users/openshift/followers",
      "following_url": "https://api.github.com/users/openshift/following{/other_user}",
      "gists_url": "https://api.github.com/users/openshift/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/openshift/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/openshift/subscriptions",
      "organizations_url": "https://api.github.com/users/openshift/orgs",
      "repos_url": "https://api.github.com/users/openshift/repos",
      "events_url": "https://api.github.com/users/openshift/events{/privacy}",
      "received_events_url": "https://api.github.com/users/openshift/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "html_url": "https://github.com/openshift/origin",
    "description": "Enterprise Kubernetes for Developers",
    "fork": false,
    "url": "https://api.github.com/repos/openshift/origin",
    "forks_url": "https://api.github.com/repos/openshift/origin/forks",
    "keys_url": "https://api.github.com/repos/openshift/origin/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/openshift/origin/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/openshift/origin/teams",
    "hooks_url": "https://api.github.com/repos/openshift/origin/hooks",
    "issue_events_url": "https://api.github.com/repos/openshift/origin/issues/events{/number}",
    "events_url": "https://api.github.com/repos/openshift/origin/events",
    "assignees_url": "https://api.github.com/repos/openshift/origin/assignees{/user}",
    "branches_url": "https://api.github.com/repos/openshift/origin/branches{/branch}",
    "tags_url": "https://api.github.com/repos/openshift/origin/tags",
    "blobs_url": "https://api.github.com/repos/openshift/origin/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/openshift/origin/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/openshift/origin/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/openshift/origin/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/openshift/origin/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/openshift/origin/languages",
    "stargazers_url": "https://api.github.com/repos/openshift/origin/stargazers",
    "contributors_url": "https://api.github.com/repos/openshift/origin/contributors",
    "subscribers_url": "https://api.github.com/repos/openshift/origin/subscribers",
    "subscription_url": "https://api.github.com/repos/openshift/origin/subscription",
    "commits_url": "https://api.github.com/repos/openshift/origin/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/openshift/origin/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/openshift/origin/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/openshift/origin/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/openshift/origin/contents/{+path}",
    "compare_url": "https://api.github.com/repos/openshift/origin/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/openshift/origin/merges",
    "archive_url": "https://api.github.com/repos/openshift/origin/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/openshift/origin/downloads",
    "issues_url": "https://api.github.com/repos/openshift/origin/issues{/number}",
    "pulls_url": "https://api.github.com/repos/openshift/origin/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/openshift/origin/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/openshift/origin/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/openshift/origin/labels{/name}",
    "releases_url": "https://api.github.com/repos/openshift/origin/releases{/id}",
    "deployments_url": "https://api.github.com/repos/openshift/origin/deployments",
    "created_at": "2014-07-30T20:04:05Z",
    "updated_at": "2019-05-20T21:45:50Z",
    "pushed_at": "2019-05-20T22:10:08Z",
    "git_url": "git://github.com/openshift/origin.git",
    "ssh_url": "git@github.com:openshift/origin.git",
    "clone_url": "https://github.com/openshift/origin.git",
    "svn_url": "https://github.com/openshift/origin",
    "homepage": "http://www.openshift.org",
    "size": 1019084,
    "stargazers_count": 6204,
    "watchers_count": 6204,
    "language": "Go",
    "has_issues": true,
    "has_projects": false,
    "has_downloads": true,
    "has_wiki": false,
    "has_pages": false,
    "forks_count": 3625,
    "mirror_url": null,
    "archived": false,
    "disabled": false,
    "open_issues_count": 517,
    "license": {
      "key": "apache-2.0",
      "name": "Apache License 2.0",
      "spdx_id": "Apache-2.0",
      "url": "https://api.github.com/licenses/apache-2.0",
      "node_id": "MDc6TGljZW5zZTI="
    },
    "forks": 3625,
    "open_issues": 517,
    "watchers": 6204,
    "default_branch": "master"
  },
  "organization": {
    "login": "openshift",
    "id": 792337,
    "node_id": "MDEyOk9yZ2FuaXphdGlvbjc5MjMzNw==",
    "url": "https://api.github.com/orgs/openshift",
    "repos_url": "https://api.github.com/orgs/openshift/repos",
    "events_url": "https://api.github.com/orgs/openshift/events",
    "hooks_url": "https://api.github.com/orgs/openshift/hooks",
    "issues_url": "https://api.github.com/orgs/openshift/issues",
    "members_url": "https://api.github.com/orgs/openshift/members{/member}",
    "public_members_url": "https://api.github.com/orgs/openshift/public_members{/member}",
    "avatar_url": "https://avatars3.githubusercontent.com/u/792337?v=4",
    "description": "The developer and operations friendly Kubernetes distro"
  },
  "sender": {
    "login": "openshift-ci-robot",
    "id": 30189103,
    "node_id": "MDQ6VXNlcjMwMTg5MTAz",
    "avatar_url": "https://avatars1.githubusercontent.com/u/30189103?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/openshift-ci-robot",
    "html_url": "https://github.com/openshift-ci-robot",
    "followers_url": "https://api.github.com/users/openshift-ci-robot/followers",
    "following_url": "https://api.github.com/users/openshift-ci-robot/following{/other_user}",
    "gists_url": "https://api.github.com/users/openshift-ci-robot/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/openshift-ci-robot/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/openshift-ci-robot/subscriptions",
    "organizations_url": "https://api.github.com/users/openshift-ci-robot/orgs",
    "repos_url": "https://api.github.com/users/openshift-ci-robot/repos",
    "events_url": "https://api.github.com/users/openshift-ci-robot/events{/privacy}",
    "received_events_url": "https://api.github.com/users/openshift-ci-robot/received_events",
    "type": "User",
    "site_admin": false
  }
}
```

Reply:
Payload I get for the issue_comment is below, which I believe in line with the GenericCommentEvent Structure. Do you want to fix the hook that handles notification?

{"author":"brahmaroutu","component":"hook","event-GUID":"79a81970-7b84-11e9-9c2f-e091254c5413","event-type":"issue_comment","level":"info","msg":"Issue comment created.","org":"ii","pr":14,"repo":"apisnoop","time":"2019-05-21T04:54:12Z","url":"https://github.com/ii/apisnoop/issues/14#issuecomment-494238935"}
{0 false created /project test_project2 need-attention https://github.com/ii/apisnoop/issues/14#issuecomment-494238935 14 {{ii   30447 https://github.com/ii} apisnoop ii/apisnoop https://github.com/ii/apisnoop true master false {false false false}} {brahmaroutu   5403596 https://github.com/brahmaroutu} {hh   31331 https://github.com/hh} [] open /woof https://github.com/ii/apisnoop/issues/14 79a81970-7b84-11e9-9c2f-e091254c5413}

Reply:
Right, but we just choose what data we get from GitHub to keep -- if you add the ID field to the structure we deserialize from the actual webhooks you'll see the data in the GenericCommentEvent